agent/engine/docker_image_manager_integ_test.go

```diff
@@ -530,24 +524,20 @@ func TestImageWithSameIDAndDifferentNames(t *testing.T) {
 
 	// Verify images are removed by docker
 	_, err = taskEngine.(*DockerTaskEngine).client.InspectImage(imageID1)
-	assert.Equal(t, docker.ErrNoSuchImage, err, "Image was not removed successfully")
+	assert.True(t, client.IsErrNotFound(err), "Image was not removed successfully")
 }
 
-// renameImage retag the image and delete the original tag
-func renameImage(original, repo, tag string, client *docker.Client) error {
-	err := client.TagImage(original, docker.TagImageOptions{
-		Repo:  repo,
-		Tag:   tag,
-		Force: false,
-	})
+// renameImage retag the image with the target tag and delete the source tag
+func renameImage(source string, target string, client *client.Client) error {
+	err := client.ImageTag(context.TODO(), source, target)
 	if err != nil {
 		return fmt.Errorf("Trying to tag image failed, err: %v", err)
 	}
 
-	// delete the original tag
-	err = client.RemoveImage(original)
+	// delete the source tag
+	_, err = client.ImageRemove(context.TODO(), source, types.ImageRemoveOptions{})
```

Comment:
Instead of directly passing in `context.TODO()`, can you do 

```
ctx, cancel := context.WithCancel(context.TODO())
defer cancel()
``` 

and pass in `ctx`?